apps/src/lib/levelbuilder/data-docs-editor/DataDocEditAll.jsx

```diff
@@ -1,11 +1,109 @@
-import React from 'react';
+import React, {useState} from 'react';
+import $ from 'jquery';
+import PropTypes from 'prop-types';
+import {Link, TextLink} from '@dsco_/link';
+import FontAwesome from '@cdo/apps/templates/FontAwesome';
+import Dialog, {Title as DialogTitle} from '@cdo/apps/templates/Dialog';
 
 const DataDocEditAll = props => {
+  const {dataDocs: initialDataDocs} = props;
+  const [dataDocs, setDataDocs] = useState(initialDataDocs);
+  const [showDeleteWarningDialog, setShowDeleteWarningDialog] = useState(false);
+  const [pendingDeleteDocKey, setPendingDeleteDocKey] = useState(null);
+
+  const initiateDeleteDataDoc = docKeyToDelete => {
+    setPendingDeleteDocKey(docKeyToDelete);
+    setShowDeleteWarningDialog(true);
+  };
+
+  const deleteDataDoc = () => {
+    $.ajax({
+      url: `/data_docs/${pendingDeleteDocKey}`,
+      method: 'DELETE',
+      success: () => {
+        setDataDocs([
+          ...dataDocs.filter(
+            dataDoc => !pendingDeleteDocKey.includes(dataDoc.key)
+          )
+        ]);
+        setShowDeleteWarningDialog(false);
+        setPendingDeleteDocKey(null);
+      },
+      error: function(xhr, ajaxOptions, thrownError) {
+        if (xhr.status === 404) {
+          setDataDocs([
+            ...dataDocs.filter(
+              dataDoc => !pendingDeleteDocKey.includes(dataDoc.key)
+            )
+          ]);
+          setShowDeleteWarningDialog(false);
+          setPendingDeleteDocKey(null);
```

Comment:
please extract duplicate code into a method - I think it meets the criteria that we'll generally want these two codepaths to stay in sync if anything changes.

Reply:
oh yes good point, thank you!